### PR TITLE
Added a key for listing all tests in a test suite.

### PR DESCRIPTION
**Pull request by therearesomewhocallmetim**
Needed for running integration tests (and possibly others) in parallel.

### CODE REVIEW

File: testing/testingmain.cpp [hunk 1]
```diff
@@ -39,6 +39,7 @@ char const kSuppressOption[] = "--suppress=";
 char const kHelpOption[] = "--help";
 char const kDataPathOptions[] = "--data_path=";
 char const kResourcePathOptions[] = "--user_resource_path=";
+char const kListAllTestsOptions[] = "--list_tests";
```
Comment (mpimenov) [inline testing/testingmain.cpp]: s/Options/Option/ ?

File: testing/testingmain.cpp [hunk 2]
```diff
@@ -144,6 +148,13 @@ int main(int argc, char * argv[])
     testResults.push_back(true);
   }
 
+  if (GetTestingOptions().m_listTests) {
```
Comment (mpimenov) [inline testing/testingmain.cpp]: {
}

File: testing/testingmain.cpp [hunk 3]
```diff
@@ -144,6 +148,13 @@ int main(int argc, char * argv[])
...
+      for (vector<string>::size_type i=0; i != testNames.size(); i++) {
```
Comment (mpimenov) [inline testing/testingmain.cpp]:
```
for (auto const & name : testNames)
  cout << name << endl;
```

**Comment by mpimenov** [comment]
LGTM

File: testing/testingmain.cpp [hunk 4]
```diff
@@ -69,6 +70,7 @@ void Usage(char const * name)
                 "Do not run tests with names corresponding to regexp.");
   DisplayOption(cerr, kDataPathOptions, "<Path>", "Path to data files.");
   DisplayOption(cerr, kResourcePathOptions, "<Path>", "Path to resources, styles and classificators.");
+  DisplayOption(cerr, kListAllTestsOptions, "List all the tests in the test suite and exit.");
```
Comment (mpimenov) [inline testing/testingmain.cpp]: s/Options/Option/

**Comment by mpimenov** [comment]
DUDE JTTP

**Comment by therearesomewhocallmetim** [comment]
DUDE JTTP

**Comment by therearesomewhocallmetim** [comment]
DUDE JTTP

**Comment by mpimenov** [comment]
s/kListAllTestsOptions/kListAllTestsOption/g

**Comment by therearesomewhocallmetim** [comment]
DUDE JTTP

File: testing/testingmain.cpp [hunk 5]
```diff
@@ -144,6 +148,13 @@ int main(int argc, char * argv[])
     testResults.push_back(true);
   }
 
+  if (GetTestingOptions().m_listTests)
+  {
+      for (auto const & name : testNames)
+          cout << name << endl;
+      return 0;
+  }
```
Comment (vng) [inline testing/testingmain.cpp]: У нас принято 2 пробела на блок везде.

**Comment by therearesomewhocallmetim** [comment]
DUDE

**Pull request merged**